### PR TITLE
Add a few tests of the PTK multiline parser

### PR DESCRIPTION
**Pull request by gforsyth**
Haven't worked much with mocks before so if this looks crazy, apologies.  

These should hopefully provide a few sanity checks if we want to refactor the multiline parser in the future.

### CODE REVIEW

File: tests/test_ptk_multiline.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""Tests sample inputs to PTK multiline and checks parser response"""
+import nose
+from nose.tools import assert_equal
+from unittest.mock import MagicMock, patch
+
+import builtins
+from xonsh.environ import Env
+builtins.__xonsh_env__ = Env()
+builtins.__xonsh_env__['INDENT'] = '    '
```
Comment (scopatz) [inline tests/test_ptk_multiline.py]: These lines could have side effects for other tests.  probably best to stick all of this stuff in a setup() function and then unwind it all in a teardown(). Same goes for below as well.

**Comment by scopatz** [comment]
This looks very nice, @gforsyth. Thanks! Only the setup and teardown issue to worry about.

**Comment by scopatz** [comment]
Hey @gforsyth, is this ready for another look? Please ping the conversation so that I get an email if it is! Thanks.

**Comment by gforsyth** [comment]
Hey @scopatz , this is ready for another check.

**Comment by scopatz** [comment]
Thanks a ton @gforsyth!

**Pull request merged**